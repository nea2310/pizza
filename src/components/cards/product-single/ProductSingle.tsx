import { LinkProps } from 'react-router-dom';
import './product-single.scss';

type TProps = {
  backUrl: string;
  image: string;
  ingredients: Array<string>;
  linkComponent: React.ForwardRefExoticComponent<LinkProps & React.RefAttributes<HTMLAnchorElement>>;
  name: string;
  price: number;
};

const ProductSingle: React.FC<TProps> = (props) => {

  const ingredients = props.ingredients.join(', ');

  return (
    <div className='product-single'>
      <h1 className='product-single__name'>{props.name}</h1>
      <div className='product-single__ingredients'>
        <strong>Состав: {ingredients}</strong>
      </div>

      <div className='product-single__img-wrapper'>
        <img className='product-single__img'
          src={props.image} alt={props.name} />
      </div>

      <div className='product-single__price'>
        <strong>Цена: {props.price}р.</strong>
      </div>

      <props.linkComponent
        className='product-single__back-link'
        to={props.backUrl}>Вернуться к меню</props.linkComponent>
    </div>
  );
}

export default ProductSingle;