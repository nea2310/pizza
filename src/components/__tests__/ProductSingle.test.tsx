import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import ProductSingle from '../cards/product-single/ProductSingle';

describe('ProductSingle component rendering', () => {
  it('Renders ProductSingle component', async () => {
    const productSingleComponent = render(
      <BrowserRouter>
        <ProductSingle
          backUrl={'/'}
          image={''}
          ingredients={['пармезан', 'моцарелла', 'камамбер']}
          name={'Маргарита'}
          price={100}
        />
      </BrowserRouter>
    );

    const user = userEvent.setup();
    expect(screen.getByText(/Маргарита/)).toBeInTheDocument();
    expect(
      screen.getByText(/Состав: пармезан, моцарелла, камамбер/)
    ).toBeInTheDocument();
    expect(screen.getByText(/Цена: 100р./)).toBeInTheDocument();
    expect(productSingleComponent).toMatchSnapshot();
  });
});
