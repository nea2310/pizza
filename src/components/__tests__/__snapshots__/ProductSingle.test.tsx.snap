// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSingle component rendering Renders ProductSingle component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="product-single"
      >
        <h1
          class="product-single__name"
        >
          Маргарита
        </h1>
        <span
          class="product-single__ingredients"
        >
          Состав: 
          пармезан, моцарелла, камамбер
        </span>
        <img
          alt="Маргарита"
          class="product-single__img"
          src=""
        />
        <span
          class="product-single__price"
        >
          Цена: 
          100
          р.
        </span>
        <a
          class="product-single__back-link"
          href="/"
        >
          Вернуться к меню
        </a>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="product-single"
    >
      <h1
        class="product-single__name"
      >
        Маргарита
      </h1>
      <span
        class="product-single__ingredients"
      >
        Состав: 
        пармезан, моцарелла, камамбер
      </span>
      <img
        alt="Маргарита"
        class="product-single__img"
        src=""
      />
      <span
        class="product-single__price"
      >
        Цена: 
        100
        р.
      </span>
      <a
        class="product-single__back-link"
        href="/"
      >
        Вернуться к меню
      </a>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
